internal/resolvers/youtube/resolver.go

```diff
@@ -51,6 +62,26 @@ func New() (resolvers []resolver.CustomURLManager) {
 		return
 	}
 
+	resolvers = append(resolvers, resolver.CustomURLManager{
+		// TODO(jammeh): handle channels where channel/user segment doesn't exist (e.g. https://www.youtube.com/tranceluv)
+		Check: func(url *url.URL) bool {
+			matches, regexErr := regexp.MatchString(`/(user|c(hannel)?)/.+`, url.Path)
+			return utils.IsSubdomainOf(url, "youtube.com") && regexErr == nil && matches
+		},
+		Run: func(url *url.URL) ([]byte, error) {
+			channelID := getYoutubeChannelIdFromURL(url)
+
+			if channelID.channelType == InvalidChannel {
+				return resolver.NoLinkInfoFound, nil
+			}
+
+			// TODO: cache key as channel/id for loading
+			channelCacheKey := constructCacheKeyFromChannelId(channelID)
+			apiResponse := channelCache.Get(channelCacheKey, nil)
+			return json.Marshal(apiResponse)
+		},
+	})
```

Comment:
Note for the future, we might want to find a way to merge this and the other resolver, similar to how Twitter resolver does that. Having three resolvers is quite overkill.
Also, `check` and `run` functions could be moved into `check.go` and `run.go` files, but we're not fully consistent with this yet, so it's fine to leave it for now as it is - just something that is really worth refactoring later.